### PR TITLE
CDAP-5194 Comprehensive list of "Known Issues"

### PR DESCRIPTION
**Pull request by awholegunch**
Consolidated all outstanding "Known Issues" under the current release notes (3.4.0). Though this list will need to be updated prior to the release (to either add or subtract issues), it is most of the work required.

**Update:** Added *Known Issues* back to all previous releases in these release notes.

[3.4.0 Release Notes, Known Issues](http://builds.cask.co/artifact/CDAP-DOB146/shared/build-6/Docs-HTML/3.4.0-SNAPSHOT/en/reference-manual/release-notes.html#known-issues)

Passes [Quick Build 5](http://builds.cask.co/browse/CDAP-DOB146-6).

Fix for [CDAP-5194](https://issues.cask.co/browse/CDAP-5194).

### CODE REVIEW

**Comment by awholegunch** [comment]
Found a few errors; fixing.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 1]
```diff
@@ -23,6 +23,143 @@ Cask Data Application Platform Release Notes
...
+- `CDAP-2945 <https://issues.cask.co/browse/CDAP-2945>`__ -
+  There is a problem in the PartitionedFileSet causing MapReduce to fail if the input
+  partition filter does not match any partitions.
+
+- `CDAP-3000 <https://issues.cask.co/browse/CDAP-3000>`__ -
+  There is a problem with the Workflow token being in an inconsistent state for nodes in a
+  fork while the fork is still running. It becomes consistent after the join.
+
+- `CDAP-3221 <https://issues.cask.co/browse/CDAP-3221>`__ -
+  There is a problem when running Standalone mode: if an adapter is configured incorrectly
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: What's an adapter?

Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: Also this is generic for all mapreduce program, irrespective of how it got created

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 2]
```diff
@@ -23,6 +23,143 @@ Cask Data Application Platform Release Notes
...
+  triggered at all. There is no warnings or messages about the missed run of the
+  workflow.
+
+- `CDAP-2945 <https://issues.cask.co/browse/CDAP-2945>`__ -
+  There is a problem in the PartitionedFileSet causing MapReduce to fail if the input
+  partition filter does not match any partitions.
+
+- `CDAP-3000 <https://issues.cask.co/browse/CDAP-3000>`__ -
+  There is a problem with the Workflow token being in an inconsistent state for nodes in a
+  fork while the fork is still running. It becomes consistent after the join.
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: It is the nodes in the fork that runs.

Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: Also, I don't like how we frame this by starting with "there is a problem". We should just say the issue upfront and not qualify it. Same is true in other places.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 3]
```diff
@@ -23,6 +23,143 @@ Cask Data Application Platform Release Notes
...
+  available.
+
+- `CDAP-1007 <https://issues.cask.co/browse/CDAP-1007>`__ -
+  If the Hive Metastore is restarted while the CDAP Explore Service is running, the
+  Explore Service remains alive, but becomes unusable. To correct, restart the CDAP Master
+  |---| which will restart all services |---| as described under "Starting CDAP Services"
+  for your particular Hadoop distribution in the :ref:`Installation documentation <installation-index>`.
+
+- `CDAP-1587 <https://issues.cask.co/browse/CDAP-1587>`__ - 
+  User datasets with names starting with ``"system"`` can potentially cause conflicts.
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: What does this mean? It coveys no information that users can use.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 4]
```diff
@@ -23,6 +23,143 @@ Cask Data Application Platform Release Notes
...
+  There is a problem with the Workflow token being in an inconsistent state for nodes in a
+  fork while the fork is still running. It becomes consistent after the join.
+
+- `CDAP-3221 <https://issues.cask.co/browse/CDAP-3221>`__ -
+  There is a problem when running Standalone mode: if an adapter is configured incorrectly
+  such that it leads to a MapReduce job to fail repeatedly, then the SDK hits an OOM exception due to ``perm gen``.
+  The Standalone needs restarting at this point.
+
+- `CDAP-3262 <https://issues.cask.co/browse/CDAP-3262>`__ -
+  There is a problem under Microsoft Windows when using the CDAP Standalone scripts with a
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: Is this is right workaround?

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 5]
```diff
@@ -23,6 +23,143 @@ Cask Data Application Platform Release Notes
...
+- `CDAP-3262 <https://issues.cask.co/browse/CDAP-3262>`__ -
+  There is a problem under Microsoft Windows when using the CDAP Standalone scripts with a  
+  JAVA_HOME that is defined as a path with spaces in it. A workaround is to use a definition of 
+  JAVA_HOME that does not include spaces.
+
+- `CDAP-3492 <https://issues.cask.co/browse/CDAP-3492>`__ -
+  In the CDAP CLI, executing ``'select *'`` from a dataset with many fields generates an error.
+  
+- `CDAP-3641 <https://issues.cask.co/browse/CDAP-3641>`__ -
+  ``WorkflowStatsSLAHttpHandler`` hangs if units are not provided.
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: This is too low level. Users who consume this don't know about SLAHttpHandler. What end point will cause this issue

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 6]
```diff
@@ -23,6 +23,143 @@ Cask Data Application Platform Release Notes
...
+
+- `CDAP-3697 <https://issues.cask.co/browse/CDAP-3697>`__ -
+  CDAP Explore is broken on secure CDH 5.1 clusters.
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: No longer applicable.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 7]
```diff
@@ -23,6 +23,143 @@ Cask Data Application Platform Release Notes
...
+  
+- `CDAP-3698 <https://issues.cask.co/browse/CDAP-3698>`__ -
+  CDAP Explore is unable to get a delegation token while fetching next results on HDP 2.0 clusters.
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: No longer applicable

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 8]
```diff
@@ -23,6 +23,143 @@ Cask Data Application Platform Release Notes
...
+  does not run on Ubuntu 14.x as of CDAP 3.0.x.
+
+- `CDAP-2721 <https://issues.cask.co/browse/CDAP-2721>`__ -
+  Metrics for :ref:`TimePartitionedFileSets <datasets-timepartitioned-fileset>` can show 
+  zero values even if there is data present.
+  
+- `CDAP-2831 <https://issues.cask.co/browse/CDAP-2831>`__ -
+  A workflow that is scheduled by time will not be run between the failure of the primary
+  master and the time that the secondary takes over. This scheduled run will not be
+  triggered at all. There is no warnings or messages about the missed run of the
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: Last line is not needed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 1 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Fixed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 2 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Fixed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 3 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Fixed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 4 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Yes.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 6 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Removed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 7 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Removed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 8 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Removed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 5 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Fixed.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 9]
```diff
@@ -23,6 +23,140 @@ Cask Data Application Platform Release Notes
...
+  were to match one of those names. To avoid this, do not start any datasets with the word
+  ``"system"``. 
+
+- `CDAP-2632 <https://issues.cask.co/browse/CDAP-2632>`__ -
+  The application in the `cdap-kafka-ingest-guide 
+  <https://github.com/cdap-guides/cdap-kafka-ingest-guide/tree/release/cdap-3.0-compatible>`__ 
+  does not run on Ubuntu 14.x as of CDAP 3.0.x.
+
+- `CDAP-2721 <https://issues.cask.co/browse/CDAP-2721>`__ -
+  Metrics for :ref:`TimePartitionedFileSets <datasets-timepartitioned-fileset>` can show
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: That's just a symptom. Known issue is different. https://issues.cask.co/browse/CDAP-2721

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 10]
```diff
@@ -23,6 +23,140 @@ Cask Data Application Platform Release Notes
...
+  Metrics for :ref:`TimePartitionedFileSets <datasets-timepartitioned-fileset>` can show 
+  zero values even if there is data present.
+  
+- `CDAP-2831 <https://issues.cask.co/browse/CDAP-2831>`__ -
+  A workflow that is scheduled by time will not be run between the failure of the primary
+  master and the time that the secondary takes over. This scheduled run will not be
+  triggered at all.
+
+- `CDAP-2945 <https://issues.cask.co/browse/CDAP-2945>`__ -
+  There is a problem in the PartitionedFileSet causing MapReduce to fail if the input
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: The feedback from earlier review was not use "There is a problem". Please fix.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 11]
```diff
@@ -23,6 +23,140 @@ Cask Data Application Platform Release Notes
...
+  The Workflow token is in an inconsistent state for nodes in a fork while the nodes of
+  the fork are still running. It becomes consistent after the join.
+
+- `CDAP-3221 <https://issues.cask.co/browse/CDAP-3221>`__ -
+  When running in CDAP Standalone mode, if a MapReduce job fails repeatedly, then the SDK
+  hits an out-of-memory exception due to ``perm gen``. The Standalone needs restarting at
+  this point.
+
+- `CDAP-3262 <https://issues.cask.co/browse/CDAP-3262>`__ -
+  For Microsoft Windows, the CDAP Standalone scripts can fail when used with a JAVA_HOME
```
Comment (sreevatsanraman) [inline cdap-docs/reference-manual/source/release-notes.rst]: This is not the right workaround. There is a different fix, please see https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/cdap-user/TQr4rtTyC0M/kje0fGdyBgAJ

**Comment by sreevatsanraman** [comment]
Took another pass. Please see comments.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 10 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Missed that one. Fixed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 9 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Fixed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 11 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Added examples of paths without spaces.

**Comment by sreevatsanraman** [comment]
LGTM

**Comment by awholegunch** [comment]
"On-hold" as I am going to move this update into the other releases of the release note page.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 12]
```diff
@@ -23,11 +23,111 @@ Cask Data Application Platform Release Notes
...
+  ``"system"``. 
+
+- `CDAP-2632 <https://issues.cask.co/browse/CDAP-2632>`__ -
+  The application in the `cdap-kafka-ingest-guide 
+  <https://github.com/cdap-guides/cdap-kafka-ingest-guide/tree/release/cdap-3.0-compatible>`__ 
+  does not run on Ubuntu 14.x as of CDAP 3.0.x.
+
+- `CDAP-2721 <https://issues.cask.co/browse/CDAP-2721>`__ -
+  Metrics for :ref:`FileSets <datasets-fileset>` can show zero values even if there is
+  data present, because FileSets do not emit metrics (`CDAP-587
```
Comment (nitinmotgi) [inline cdap-docs/reference-manual/source/release-notes.rst]: CDAP-2721 line doesn't represent hyperlink correctly for JIRA.

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 13]
```diff
@@ -23,11 +23,111 @@ Cask Data Application Platform Release Notes
...
+  
+- `CDAP-3641 <https://issues.cask.co/browse/CDAP-3641>`__ -
+  A RESTful API call to retrieve workflow statistics hangs if units (such as "s" for
+  seconds) are not provided as part of the query.
+
+- `CDAP-3750 <https://issues.cask.co/browse/CDAP-3750>`__ -
+  If a table schema contains a field name that is a reserved word in the Hive DDL, ``'enable explore'`` fails.
+
+
 `Release 3.3.2 <http://docs.cask.co/cdap/3.3.2/index.html>`__
```
Comment (nitinmotgi) [inline cdap-docs/reference-manual/source/release-notes.rst]: Why is there a link to 3.3.2 here ?

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 12 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Fixed.

File: cdap-docs/reference-manual/source/release-notes.rst (hunk 13 above)
Comment (awholegunch) [inline cdap-docs/reference-manual/source/release-notes.rst]: Because that is the previous set of documentation to the 3.4.0 set.

**Comment by sreevatsanraman** [comment]
LGTM

**Pull request merged**